cluster/kubernetes/resource/load.go

```diff
@@ -18,16 +18,24 @@ import (
 func Load(base, atLeastOne string, more ...string) (map[string]resource.Resource, error) {
 	roots := append([]string{atLeastOne}, more...)
 	objs := map[string]resource.Resource{}
+	charts, err := newChartTracker(base)
+	if err != nil {
+		return nil, errors.Wrapf(err, "walking %q for chartdirs", base)
+	}
 	for _, root := range roots {
 		err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
 			if err != nil {
 				return errors.Wrapf(err, "walking %q for yamels", path)
 			}
 
-			if info.IsDir() && looksLikeChart(path) {
+			if charts.isChart(path) {
```

Comment:
`isChart` and `inChart` are a little too similar -- I can imagine typo-flips between them. How about `isDirChart` and `isPathInChart`, or similar?